docs/tutorials/control/air-quality-fleet.md

```diff
@@ -0,0 +1,953 @@
+           PM25: entry.data.readings["pm_2.5"],
+           time: entry.time_received,
+         });
+       }
+     }
+
+     // Sort the air quality data from this machine
+     // by timestamp
+     thedata = thedata.sort(function (a, b) {
+       let x = a.time.toString();
+       let y = b.time.toString();
+       if (x < y) {
+         return -1;
+       }
+       if (x > y) {
+         return 1;
+       }
+       return 0;
+     });
+
+     // Add up the last 5 readings collected.
+     // If there are fewer than 5 readings, add all of them.
+     let x = 5; // The number of readings to average over
+     if (x > thedata.length) {
+       x = thedata.length;
+     }
+     let total = 0;
+     for (let i = 1; i <= x; i++) {
+       const reading: number = thedata[thedata.length - i].PM25;
+       total += reading;
+     }
+     // Return the average of the last few readings
+     return total / x;
+   }
+   ```
+
+1. Now that you've defined the function to sort and average the data for each machine, you're done with all the `dataClient` code.
+   The final piece you need to add to this script is a way to create some HTML to display data from each machine in your dashboard.
+
+   Paste the following code into the main function of <file>main.ts</file>, in place of `// <Insert HTML block code here in later steps>`:
+
+   ```typescript {class="line-numbers linkable-line-numbers"}
+   // Instantiate the HTML block that will be returned
+   // once everything is appended to it
+   let htmlblock: HTMLElement = document.createElement("div");
+
+   // Display the relevant data from each machine to the dashboard
+   for (const mach of machineIDs) {
+     let insideDiv: HTMLElement = document.createElement("div");
+     let avgPM: number = await getLastFewAv(thedata, mach._id);
+     // Color-code the dashboard based on air quality category
+     let level: string = "blue";
+     switch (true) {
+       case avgPM < 12.1: {
+         level = "good";
+         break;
+       }
+       case avgPM < 35.5: {
+         level = "moderate";
+         break;
+       }
+       case avgPM < 55.5: {
+         level = "unhealthy-sensitive";
+         break;
+       }
+       case avgPM < 150.5: {
+         level = "unhealthy";
+         break;
+       }
+       case avgPM < 250.5: {
+         level = "very-unhealthy";
+         break;
+       }
+       case avgPM >= 250.5: {
+         level = "hazardous";
+         break;
+       }
+     }
+     // Create the HTML output for this machine
+     insideDiv.className = "inner-div " + level;
+     insideDiv.innerHTML =
+       "<p>" +
+       mach._id +
+       ": " +
+       avgPM.toFixed(2).toString() +
+       " &mu;g/m<sup>3</sup></p>";
+     htmlblock.appendChild(insideDiv);
+   }
+
+   // Output a block of HTML with color-coded boxes for each machine
+   return document.getElementById("insert-readings").replaceWith(htmlblock);
+   ```
+
+{{%expand "Click to see the full TypeScript code" %}}
```

Comment:
this needs to have it's own header and be linked to prominently, so that people don't miss it

Reply:
Ooh oops I could've sworn I put it in. Doing now!

Reply:
I think add a line here with a link to #full-tutorial-code